sky/clouds/service_catalog/data_fetchers/fetch_aws.py

```diff
@@ -320,18 +334,35 @@ def fetch_availability_zone_mappings() -> pd.DataFrame:
     parser.add_argument('--no-az-mappings',
                         dest='az_mappings',
                         action='store_false')
+    parser.add_argument('--check-regions-integrity',
+                        action='store_true',
+                        help='Check the region integrity of the fetched data.')
     parser.set_defaults(az_mappings=True)
     args = parser.parse_args()
 
     region_filter = ALL_REGIONS if args.all_regions else US_REGIONS
 
+    def _check_regions_integrity(df: pd.DataFrame, name: str):
+        fetched_regions = set(df['Region'].unique())
+        requested_regions = set(region_filter)
+        if fetched_regions != requested_regions:
```

Comment:
When will this happen? Worth a comment?

Reply:
It can happen if we fail to fetch any of the regions in `region_filter`, due to network issues, AWS API glitch, etc.

Reply:
I see. If users manually run this fetcher, is it likely such errors happen to them (I've seen some IncompleteRead)? If so, is not checking okay? Asking as I'm thinking whether this flag should be removed and always default to yes.

Reply:
Good point! I changed logic of the fetch_aws:
1. We check the fetched regions contains all the regions enabled with the user's account by default
2. We always fetch all the regions, instead of the US regions
3. If `--strict-all-regions` is specified, we will check if the fetched regions match the `ALL_REGIONS` list despite the enabled regions of the user's account.
PTAL.